plugins/newsdownloader.koplugin/main.lua

```diff
@@ -322,39 +322,46 @@ local function getTitleWithDate(feed)
 end
 
 function NewsDownloader:downloadFeed(feed, feed_output_dir, include_images, message)
-    local link = getFeedLink(feed.link)
     local title_with_date = getTitleWithDate(feed)
-    local news_dl_path = ("%s%s%s"):format(feed_output_dir,
-                                           title_with_date,
-                                           file_extension)
+    local news_file_path = ("%s%s%s"):format(feed_output_dir,
+                                             title_with_date,
+                                             file_extension)
 
-    local file_mode = lfs.attributes(news_dl_path, "mode")
+    local file_mode = lfs.attributes(news_file_path, "mode")
     if file_mode == "file" then
-        logger.dbg("NewsDownloader:", news_dl_path, "already exists. Skipping")
+        logger.dbg("NewsDownloader:", news_file_path, "already exists. Skipping")
     else
-        logger.dbg("NewsDownloader: News file will be stored to :", news_dl_path)
+        logger.dbg("NewsDownloader: News file will be stored to :", news_file_path)
         local article_message = T(_("%1\n%2"), message, title_with_date)
-        DownloadBackend:download(link, news_dl_path, include_images, article_message)
+        local link = getFeedLink(feed.link)
+        local html = DownloadBackend:loadPage(link)
+        DownloadBackend:createEpub(news_file_path, html, link, include_images, article_message)
     end
 end
 
-function NewsDownloader:createFromDescription(feed, context, feed_output_dir)
+function NewsDownloader:createFromDescription(feed, context, feed_output_dir, include_images, message)
```

Comment:
The bulk of `NewsDownloader:downloadFeed` and `NewsDownloader:createFromDescription` is now identical.